### PR TITLE
test: first rtl test

### PR DESCRIPTION
**Pull request by haxxmaxx**
Setting up RTL and adding test for `TabelWrapper`. The big question mark is if the `test-setup` approach is solid of if there should be more/less in the script. WIll cover other components in next PR -> get coverage back up

Also some renaming and small refactor of `withSlections`

### CODE REVIEW

File: src/table/__tests__/TableWrapper.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    rowsPerPage = 100;
+  });
+
+  beforeEach(() => {
+    sandbox.verifyAndRestore();
+    sandbox.resetHistory();
+  });
+
+  it('should render table', () => {
+    mockContent();
```
Comment (haxxmaxx) [inline src/table/__tests__/TableWrapper.spec.jsx:39]: @niekvanstaveren Can't seem to run this in `beforeEach`, don't see why? Ideas?

File: src/table/__tests__/TableWrapper.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    const { queryByLabelText, queryByText } = render(
+      <TableWrapper tableData={tableData} setPageInfo={setPageInfo} constraints={constraints} />
+    );
+
+    expect(queryByLabelText('sticky table')).to.be.visible;
+    expect(queryByText(`1-${rowsPerPage} of ${tableData.size.qcy}`)).to.be.visible;
+    expect(queryByText(rowsPerPage)).to.be.visible;
```
Comment (haxxmaxx) [inline src/table/__tests__/TableWrapper.spec.jsx:47]: There are a couple of other strings to look at, but this kind of what we control, the rest is Mui. Enough?

File: src/table/__tests__/TableWrapper.spec.jsx (hunk 1 above)
Comment (niekvanstaveren) [reply src/table/__tests__/TableWrapper.spec.jsx:39]: You have two beforeEach defined, shouldn't the second one be an afterEach? Not sure if that fixes your issue.....

Comment (haxxmaxx) [reply src/table/__tests__/TableWrapper.spec.jsx:39]: of course...

File: src/table/TableHeadWrapper.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+          <TableCell key={column.id} align={column.align} style={{ minWidth: column.minWidth }}>
+            {column.label}
+          </TableCell>
+        ))}
+      </TableRow>
+    </TableHead>);
+}
+
+TableHeadWrapper.propTypes = {
+  columns: PropTypes.array.isRequired,
```
Comment (niekvanstaveren) [inline src/table/TableHeadWrapper.jsx:23]: Is this correct? Seems columns is not a direct property

File: src/table/TableWrapper.jsx [hunk 4]
```diff
@@ -3,12 +3,10 @@ import PropTypes from 'prop-types';
...
 import Paper from '@material-ui/core/Paper';
 import Table from '@material-ui/core/Table';
-import TableCell from '@material-ui/core/TableCell';
 import TableContainer from '@material-ui/core/TableContainer';
-import TableHead from '@material-ui/core/TableHead';
 import TablePagination from '@material-ui/core/TablePagination';
-import TableRow from '@material-ui/core/TableRow';
-import TableBodyWrapper from './tableBody';
+import TableBodyWrapper from './TableHeadWrapper';
+import TableHeadWrapper from './TableBodyWrapper';
```
Comment (niekvanstaveren) [inline src/table/TableWrapper.jsx:9]: ????

File: src/table/__tests__/TableWrapper.spec.jsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  it('should change back to first page when not on first page and no rows', async () => {
+    tableData.rows = [];
+
+    const { findByTitle } = render(
+      <TableWrapper tableData={tableData} setPageInfo={setPageInfo} constraints={constraints} />
+    );
+    fireEvent.click(await findByTitle('Next page'));
+
+    // Called when pressing the button
+    expect(setPageInfo).to.have.been.calledWith({ top: rowsPerPage, height: rowsPerPage });
```
Comment (niekvanstaveren) [inline src/table/__tests__/TableWrapper.spec.jsx:65]: Interesting that this is still called, should the button not be inactive?

**Review by niekvanstaveren — APPROVED** [review]
Looks good

File: src/table/__tests__/TableWrapper.spec.jsx (hunk 5 above)
Comment (haxxmaxx) [reply src/table/__tests__/TableWrapper.spec.jsx:63]: Yeah this pattern is weird. i just looked at it. the pagination does not look at `rows`, only `size.qcy`. I think I should do a somewhat realistic flow here

File: src/table/TableWrapper.jsx (hunk 4 above)
Comment (haxxmaxx) [reply src/table/TableWrapper.jsx:9]: hahaha

File: src/table/TableHeadWrapper.jsx (hunk 3 above)
Comment (haxxmaxx) [reply src/table/TableHeadWrapper.jsx:23]: This is incorrect, I'll fix

File: src/table/__tests__/TableWrapper.spec.jsx (hunk 5 above)
Comment (haxxmaxx) [reply src/table/__tests__/TableWrapper.spec.jsx:63]: Actually, I think it is fine. It is a hacky way to simulate the real use case: that you confirm selections on any page past the first page and you don't select more rows than rows per page. The data paging then only fetches rows 101-200 or something when there are less than 100 rows, which returns 0 rows.

**Pull request merged**